### PR TITLE
Tweak wrapper

### PR DESCRIPTION
**Pull request by jfear**
Changing some of the things we talked about.

### CODE REVIEW

File: include/WRAPPER_SLURM [hunk 1]
```diff
@@ -1,28 +1,47 @@
 #!/bin/bash
+#SBATCH --job-name="lcdb-wf"
+#SBATCH --partition="niddk,norm"
+#SBATCH --time=24:00:00
 
-if [ -z $SNAKEFILE ]; then
-    SNAKEFILE=$1
-fi
```
Comment (jfear) [inline include/WRAPPER_SLURM]: did not find this to be needed, we can add it back but `$1` should always be the snakefile.

File: include/WRAPPER_SLURM [hunk 2]
```diff
@@ -1,28 +1,47 @@
...
-if [ -z $EXTRA ]; then
+if [ -z $2 ]; then
     EXTRA=$2
+else
+    EXTRA=''
 fi
 
 
+# make logdir
+if [[ ! -e logs ]]; then mkdir -p logs; fi
```
Comment (jfear) [inline include/WRAPPER_SLURM]: Make a logs dir if needed.

File: include/WRAPPER_SLURM [hunk 3]
```diff
@@ -1,28 +1,47 @@
...
     -T \
+    -k \
     --rerun-incomplete \
     --jobname "s.{rulename}.{jobid}.sh" \
     -j 999 \
-    --cluster-config clusterconfig.yaml \
+    --cluster-config config/clusterconfig.yaml \
     --verbose \
-    --cluster 'sbatch {cluster.prefix} --cpus-per-task={threads} ' \
+    --cluster 'sbatch {cluster.prefix} --cpus-per-task={threads}  --output=logs/{rule}.o.%j --error=logs/{rule}.e.%j' \
```
Comment (jfear) [inline include/WRAPPER_SLURM]: I tried adding `{jobid}` to the log file name, but in the version of snakemake, `jobid` is not accessible so it does not work.

File: include/WRAPPER_SLURM [hunk 4]
```diff
@@ -1,28 +1,47 @@
...
     --use-conda \
-    --configfile config.yml \
+    --configfile config/test_config.yaml \
```
Comment (jfear) [inline include/WRAPPER_SLURM]: Oops, I had this in for testing, need to revert

File: include/WRAPPER_SLURM [hunk 5]
```diff
@@ -1,28 +1,47 @@
...
+SNAKE_PID=$!
+
+finish(){
+    echo 'Stopping running snakemake job.'
+    kill -SIGINT $SNAKE_PID
+    exit 0
+}
+trap finish SIGTERM
+
+wait $SNAKE_PID
```
Comment (jfear) [inline include/WRAPPER_SLURM]: I add this logic so if you need to `scancel` it should capture the signal and send a keyboard interrupt to slurm. Hopefully this helps snakemake shutdown a little cleaner and keep things unlocked.

File: include/WRAPPER_SLURM [hunk 6]
```diff
@@ -1,28 +1,47 @@
...
     --jobname "s.{rulename}.{jobid}.sh" \
     -j 999 \
-    --cluster-config clusterconfig.yaml \
+    --cluster-config config/clusterconfig.yaml \
     --verbose \
-    --cluster 'sbatch {cluster.prefix} --cpus-per-task={threads} ' \
+    --cluster 'sbatch {cluster.prefix} --cpus-per-task={threads}  --output=logs/{rule}.o.%j --error=logs/{rule}.e.%j' \
     --use-conda \
-    --configfile config.yml \
+    --configfile config/config.yml \
```
Comment (jfear) [inline include/WRAPPER_SLURM]: Probably should standardize yaml names. Which do you prefer `.yaml` or `.yml`?

File: include/WRAPPER_SLURM (hunk 1 above)
Comment (daler) [inline include/WRAPPER_SLURM]: sounds good

File: include/WRAPPER_SLURM (hunk 3 above)
Comment (daler) [inline include/WRAPPER_SLURM]: bummer!

File: include/WRAPPER_SLURM (hunk 5 above)
Comment (daler) [inline include/WRAPPER_SLURM]: maybe exit 1?

File: include/WRAPPER_SLURM (hunk 6 above)
Comment (daler) [inline include/WRAPPER_SLURM]: No preference. I guess snakemake wrappers expect `environment.yaml`, so let's arbitrarily go with that.

**Pull request merged**